packages/runtime/container-runtime/src/summarizerHeuristics.ts

```diff
@@ -0,0 +1,95 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Timer } from "@fluidframework/common-utils";
+import { ISummaryConfiguration } from "@fluidframework/protocol-definitions";
+import { ISummarizeHeuristicData, ISummarizeHeuristicRunner, ISummaryAttempt } from "./summarizerTypes";
+import { SummarizeReason } from "./summaryGenerator";
+
+/** Simple implementation of class for tracking summarize heuristic data. */
+export class SummarizeHeuristicData implements ISummarizeHeuristicData {
+    protected _lastAttempt: ISummaryAttempt;
+    public get lastAttempt(): ISummaryAttempt {
+        return this._lastAttempt;
+    }
+
+    protected _lastSuccessfulSummary: Readonly<ISummaryAttempt>;
+    public get lastSuccessfulSummary(): Readonly<ISummaryAttempt> {
+        return this._lastSuccessfulSummary;
+    }
+
+    constructor(
+        public lastOpSequenceNumber: number,
+        attemptBaseline: ISummaryAttempt,
+    ) {
+        this._lastAttempt = attemptBaseline;
+        this._lastSuccessfulSummary = { ...attemptBaseline };
+    }
+
+    public initialize(lastSummary: Readonly<ISummaryAttempt>) {
+        this._lastAttempt = lastSummary;
+        this._lastSuccessfulSummary = { ...lastSummary };
+    }
+
+    public recordAttempt(refSequenceNumber?: number) {
+        this._lastAttempt = {
+            refSequenceNumber: refSequenceNumber ?? this.lastOpSequenceNumber,
+            summaryTime: Date.now(),
+        };
+    }
+
+    public markLastAttemptAsSuccessful() {
+        this._lastSuccessfulSummary = { ...this.lastAttempt };
+    }
+}
+
+/**
+ * This class contains the heuristics for when to summarize.
+ */
+export class SummarizeHeuristicRunner implements ISummarizeHeuristicRunner {
+    private readonly idleTimer: Timer;
+
+    public constructor(
+        private readonly heuristicData: ISummarizeHeuristicData,
+        private readonly configuration: ISummaryConfiguration,
+        private readonly trySummarize: (reason: SummarizeReason) => void,
+        private readonly minOpsForAttemptOnClose = 50,
+    ) {
+        this.idleTimer = new Timer(
+            this.configuration.idleTime,
+            () => this.trySummarize("idle"));
+    }
+
+    public countOpsSinceLastAck(): number {
```

Comment:
nit: this can just be `opsSinceLastAck`. Count makes me think that this method will count the number of ops but not return it :).

Reply:
Agree with this, I actually had the same thought.. it feels like it's doing a loop or something by reading that.

Reply:
Changed to getter.